packages/commerce-server/src/providers/default-payment-options.ts

```diff
@@ -1,12 +1,50 @@
 import Stripe from 'stripe'
+import {z} from 'zod'
 
 type StripeConfig = {
   stripeSecretKey: string
   apiVersion: '2020-08-27'
 }
 
-type StripeProvider = {name: 'stripe'; paymentClient: Stripe}
-type StripeProviderFunction = (options: StripeConfig) => StripeProvider
+export const PurchaseMetadata = z.object({
+  country: z.string().optional(),
+  appliedPPPStripeCouponId: z.string().optional(), // TODO: make this provider agnostic
+  upgradedFromPurchaseId: z.string().optional(),
+  usedCouponId: z.string().optional(),
+})
+
+export const PurchaseInfoSchema = z.object({
+  customerIdentifier: z.string(),
+  email: z.string().nullable(),
+  name: z.string().nullable(),
+  productIdentifier: z.string(),
+  product: z.object({name: z.string().nullable()}), // TODO: does this need to surface any other values?
+  chargeIdentifier: z.string(),
+  couponIdentifier: z.string().optional(),
+  quantity: z.number(),
+  chargeAmount: z.number(),
+  metadata: PurchaseMetadata.passthrough().optional(),
+})
+export type PurchaseInfo = z.infer<typeof PurchaseInfoSchema>
+
+type PaymentProviderFunctionality = {
+  getPurchaseInfo: (checkoutSessionId: string) => Promise<PurchaseInfo>
+}
```

Comment:
the first piece of _payment functionality_ moving into the umbrella of payment providers, additional pieces will be added to this type which the different payment providers conform to